cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+)
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.lib.string_diff import get_cosine_similarity
+from cl.people_db.lookup_utils import (
+    find_all_judges,
+    find_just_name,
+    lookup_judges_by_last_name_list,
+)
+from cl.people_db.models import Person
+from cl.search.models import SOURCES, Docket, Opinion, OpinionCluster
+
+VALID_UPDATED_DOCKET_SOURCES = [
+    Docket.COLUMBIA,
+    Docket.COLUMBIA_AND_RECAP,
+    Docket.COLUMBIA_AND_SCRAPER,
+    Docket.COLUMBIA_AND_RECAP_AND_SCRAPER,
+    Docket.COLUMBIA_AND_IDB,
+    Docket.COLUMBIA_AND_RECAP_AND_IDB,
+    Docket.COLUMBIA_AND_SCRAPER_AND_IDB,
+    Docket.COLUMBIA_AND_RECAP_AND_SCRAPER_AND_IDB,
+    Docket.HARVARD_AND_COLUMBIA,
+    Docket.COLUMBIA_AND_RECAP_AND_HARVARD,
+    Docket.COLUMBIA_AND_SCRAPER_AND_HARVARD,
+    Docket.COLUMBIA_AND_RECAP_AND_SCRAPER_AND_HARVARD,
+    Docket.COLUMBIA_AND_IDB_AND_HARVARD,
+    Docket.COLUMBIA_AND_RECAP_AND_IDB_AND_HARVARD,
+    Docket.COLUMBIA_AND_SCRAPER_AND_IDB_AND_HARVARD,
+    Docket.COLUMBIA_AND_RECAP_AND_SCRAPER_AND_IDB_AND_HARVARD,
+]
+
+
+VALID_MERGED_SOURCES = [
+    key
+    for key in dict(SOURCES.NAMES).keys()
+    if SOURCES.COLUMBIA_ARCHIVE in key
+]
+
+
+def clean_opinion_content(content: str, is_harvard: bool) -> str:
+    """Strip all non-alphanumeric characters
+    :param content: content from opinion
+    :param is_harvard: true if content is from harvard
+    :return: cleaned content
+    """
+    soup = BeautifulSoup(content, features="html.parser")
+
+    if is_harvard:
+        for op in soup.select("opinion"):
+            # Remove any author tag inside opinion
+            for extra in op.find_all(["author"]):
+                extra.extract()
+
+    prep_text = re.sub(
+        r"[^a-zA-Z0-9 ]", "", soup.getText(separator=" ").lower()
+    )
+    prep_text = re.sub(" +", " ", prep_text)
+    return prep_text
+
+
+def get_cl_opinion_content(cluster_id: int) -> list[dict[Any, Any]]:
+    """Get the opinions content for a cluster object
+    :param cluster_id: Cluster ID for a set of opinions
+    :return: list with opinion content from cl
+    """
+    cl_cleaned_opinions = []
+    opinions_from_cluster = Opinion.objects.filter(cluster_id=cluster_id)
+    is_harvard = False
+
+    for i, op in enumerate(opinions_from_cluster):
+        content = None
+        if len(op.xml_harvard) > 1:
+            content = op.xml_harvard
+            is_harvard = True
+        elif len(op.html_with_citations) > 1:
+            content = op.html_with_citations
+        elif len(op.html_columbia) > 1:
+            content = op.html_columbia
+        elif len(op.html_lawbox) > 1:
+            content = op.html_lawbox
+        elif len(op.plain_text) > 1:
+            content = op.plain_text
+        elif len(op.html) > 1:
+            content = op.html
+
+        if not content:
+            raise EmptyOpinionException("No content in opinion")
+
+        prep_text = clean_opinion_content(content, is_harvard=is_harvard)
```

Comment:
why do we throw an error here - if no content in an opinion - there can be another opinion that could be used to match something up?  also it seems really unlikely no?

Reply:
yes, it seems really unlikely, i'll remove it